identity/handler.go

```diff
@@ -540,3 +545,86 @@ func (h *Handler) delete(w http.ResponseWriter, r *http.Request, ps httprouter.P
 
 	w.WriteHeader(http.StatusNoContent)
 }
+
+// swagger:parameters adminPatchIdentity
+// nolint:deadcode,unused
+type adminPatchIdentity struct {
+	// ID must be set to the ID of identity you want to update
+	//
+	// required: true
+	// in: path
+	ID string `json:"id"`
+
+	// in: body
+	Body jsonpatch.Patch
```

Comment:
Unfortunately, this does no generated a correct OpenAPI definition (check the `spec/api.json` file). The problem is that we're using an external library here. This does not know go-swagger (the tool we use to extract the OpenAPI Spec from Go Code) and does not have the correct struct annotations.

To fix this, include this struct instead: https://github.com/ory/x/blob/463cd057b17c6d312dd26dd22987369d7c721c6b/openapix/jsonpatch.go#L6

Please also note that JSON Patch APIs usually allow us to send an array of patches, not just one patch :) That's why we want to use JSONPatchDocument instead!

There's one more thing though. The problem is that go-swagger is incorrectly tagging struct fields that are `interface{}`. To correctly handle the type in OpenAPI, you will need to add a JSON Patch for the OpenAPI spec, like we do in Ory Hydra:

https://github.com/ory/hydra/blob/e3c319eb1b1d62b44139dd583392f4ac9bf8ecc7/.schema/openapi/patches/common.yaml#L1-L2

Lastly, you'll need to add the `github.com/ory/x/openapix` package to the paths to resolve in the Makefile, like we do in Ory Hydra: 

https://github.com/ory/hydra/blob/e3c319eb1b1d62b44139dd583392f4ac9bf8ecc7/Makefile#L112

The API tooling is currently very fiddely. I hope that we can improve this some day!

Reply:
Thanks for the write-up. I added the `jsonpatch.Patch` struct as a placeholder & then forgot to check what the output is. The Patch body should now generate correctly.

Unfortunately, I now get some seemingly unrelated changes in the `Pagination` & `PaginationHeaders` files when generating with the same setup as you have in hydra. I am not sure what the issue is and could not figure out how they relate to my changes. Could you have a look at that and let me know if there is an issue?

Reply:
damn :/ that is probably because of https://github.com/ory/x/blob/463cd057b17c6d312dd26dd22987369d7c721c6b/openapix/pagination.go#L3-L42 which I recently changed. I'll fix that and release a new ory/x version

Reply:
https://github.com/ory/x/compare/token-pagination-naming?expand=1

Reply:
Updating to ory x [v0.0.439](https://github.com/ory/x/releases/tag/v0.0.439) should fix this problem

Reply:
I used the newest version of `ory/x`. I am still generating unrelated models, but I guess that is intended?
Might want to that in a separate commit to master instead though. I can open a separate PR to update ory/x and generate the necessary models, if needed.

Reply:
The dependency updates seem to have a broken atleast the test `TestNewRecoveryEmailAddress` in  `identity/identity_recovery_test.go` as a [UUID consisting of only zeroes is now considered empty](https://github.com/stretchr/testify/blob/master/assert/assertions.go#L579), which makes the assertion fail. I fixed the test here, but I still think this should be done in a separate PR. Let me know how you do those kind of things here :)